### ISSUE TITLE
Golang isn't required

### ISSUE THREAD

**Issue by nacho00112** [KJHJason/Cultured-Downloader-CLI#29]
No matter if python or golang it won't affect the download speed, the matter is the startup time which could be slightly different, i don't which one further than the other, since i don't know what stuff golang loads.

About the concurrency, python have many ways to do concurrency, asyncio, threading etc.

Python is better in terms that it does not require compiling, then is less work to the user, specially if he don't have golang installed

**Comment by KJHJason** [comment]
Download speeds are not really affected by the programming language but mainly by your network speed. However, It's up to one's personal preferences, I decided to port the original Python program to Golang when I first started learning Golang as a challenge.

However, the main factor that matters the most to me is the fact that it's a statically typed language. It is obvious that Python is the go-to for automation but if there's a need to maintain, more effort is needed for Python due to its dynamically typed nature which would require type hinting. Additionally, since we are dealing with APIs, dataclasses or pydantic inherited classes would be needed for JSON schema and JSON validations which also take time to implement (mainly the JSON validations).

Regarding Python's concurrency, asyncio is the most preferred over threading as Python's threading will take longer to spawn and will accumulate more memory as more threads are spawned than languages like Exilir and Golang. Additionally, KeyboardInterrupt error handling on Python's threading is very tedious to handle unlike Golang's since Golang is able to communicate through channels and contexts to signal each Goroutines to stop.
That said, asynchronous codes in Python can still be blocked, unlike Goroutines.

tldr:
Was learning Golang hence I ported it to Golang.
Preferred it as it is a statically typed language.
Concurrency is still way better than Python's.
